### PR TITLE
Finished Assignment 2

### PR DESCRIPTION
**Pull request by CodyGarciaa**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- This PR is my finished assignment 2
- key changes: post is rendered using data fetched from the supabase instead of hard coded
- because there is more posts in the supabase now, had to select the one 'rbeggs' post manually
- converted supabase date format to viewable "Month Day" format

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
IMG_23A33D1001DA-1

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- there is only one file to review: PostScreen.tsx
- this should be reviewed all at once, have few commits just after finishing each part of the assignment
- no userflows or edge cases to test for

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ronniebeggs

### CODE REVIEW

File: src/screens/PostScreen.tsx [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
-  //   imageURL: 'imageURL',
-  //   likes: 0,
-  // });
+  const [postData, setPostData] = useState({
+    created_at: 'created_at',
+    description: 'description',
+    image_url: 'image_url',
+    like_count: 0,
+    username: 'username',
+    uuid: 'uuid',
```
Comment (ronniebeggs) [inline src/screens/PostScreen.tsx:16]: I'm assuming these are placeholders that appear while the supabase query loads? If so, we can instead create a `isLoading` or `postData == null` condition that only renders `postData` after the data is fetched. You can use an inline if condition (`<condition> ? <then-body> : <else-body>`) to render either the postData or a loading screen.

Comment (ronniebeggs) [reply src/screens/PostScreen.tsx:16]: Additionally, I'd consider creating a `Post` type so these fields and their corresponding types are explicitly declared.

File: src/screens/PostScreen.tsx [hunk 2]
```diff
@@ -36,36 +57,27 @@ export default function PostScreen() {
...
-          training programs, and mental health support. read more online:
-          <Text style={styles.link}>https://www.codetenderloin.org/</Text>
-        </Text>
+        </Pressable>
+        <Text style={styles.caption}>{postData['description']}</Text>
         <Image
           style={styles.postImage}
           source={{
-            uri: 'https://cdn.britannica.com/51/178051-050-3B786A55/San-Francisco.jpg',
+            uri: postData['image_url'],
```
Comment (ronniebeggs) [inline src/screens/PostScreen.tsx:73]: I typically use the `postData.image_url` dot syntax for accessing object attributes. Use whatever is consistent with the rest of your project.

File: src/supabase/client.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// if (
+//   !process.env.EXPO_PUBLIC_SUPABASE_URL ||
+//   !process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY
+// ) {
+//   throw new Error('Supabase environment variables are not defined.');
+// }
+// const supabase = createClient(
+//   process.env.EXPO_PUBLIC_SUPABASE_URL,
+//   process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY,
+// );
```
Comment (ronniebeggs) [inline src/supabase/client.ts:13]: Remove commented code before committing!

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
I left a few minor comments but otherwise this looks really good! Thanks also for writing such a detailed PR description. Good job!
